lib/install/management/delete.go

```diff
@@ -238,9 +238,33 @@ func (d *Dispatcher) DeleteVCHInstances(vmm *vm.VirtualMachine, conf *config.Vir
 	var err error
 	var children []*vm.VirtualMachine
 
-	// TODO: GET CHILDREN EITHER BY FOLDER OR RP BASED ON DEPLOYMENT
-	if children, err = d.parentResourcepool.GetChildrenVMs(d.op, d.session); err != nil {
-		return err
+	folderRef, err := vchFolder(d.op, d.session, conf)
+
+	if err != nil || folderRef == nil {
```

Comment:
I think the logic is to loose here...at a minimum you need to include whether DRS is enabled -- if it's a DRS disabled cluster and the `vchFolder` call fails...this will delete all containerVMs in the cluster..not good

Reply:
True, not good at all... We might need a way to force this in the future. I can think of a way to test this using VCSIM. As for how to mitigate this... This case is valid when in a DRS enabled. So I am pushing an addition which checks the DRSEnabled inside of this branching logic. Returning a new error message in that case. @anchal-agrawal @cgtexmex please review my error messaging and let me know if it is up to par with respect to clarity and handling.